src/v/coproc/event_listener.cc

```diff
@@ -80,17 +88,69 @@ event_listener::persist_actions(absl::btree_map<script_id, iobuf> wsas) {
 }
 
 event_listener::event_listener(ss::sharded<pacemaker>& pacemaker)
-  : _client({config::shard_local_cfg().kafka_api()[0].address})
-  , _dispatcher(pacemaker, _abort_source) {}
+  : _dispatcher(pacemaker, _abort_source) {}
+
+ss::future<> event_listener::do_connect() {
+    vassert(!_client, "do_connect() called when theres already a valid client");
+    /// Invoke metadata request on a new client connected to this broker
+    auto client = std::make_unique<kafka::client::transport>(
+      rpc::base_transport::configuration{
+        .server_addr = ss::socket_address(
+          ss::net::ipv4_address("127.0.0.1"), 9092)});
```

Comment:
Should it? What value does it offer to be configurable? Also, here we are working under the assumption that there is a kafka node available at localhost 9092 because we are running on that node. So we can 100% dial to it and obtain metadata, unless there is some error

Reply:
But user can use port 1000 or other, it has to be configurable.

Reply:
Admittedly changing the port would break this, makes sense